website/src/controllers/groups.js

```diff
@@ -137,7 +137,25 @@ api.get = function(req, res, next) {
   q.exec(function(err, group){
     if (err) return next(err);
     if (!group && gid!=='party') return res.json(404,{err: "Group not found or you don't have access."});
-    res.json(group);
+
+    //Since we have a limit on how many members are populate to the group, we want to make sure the user is always in the group
+    var userInGroup = _.find(group.members, function(member){ return member._id == user._id; });
+
+    //If the group is private or the group is a party, then the user must be a member of the group based on access restrictions above
+    if (group.privacy === 'private' || gid === 'party') {
+      //If the user is not in the group query, add them
+      if (userInGroup === undefined) { group.members.push(user); }
+      res.json(group);
+    } else if ( group.privacy === "public" ) { //The group is public, we must do an extra check to see if the user is already in the group query
+      //We must see how to check if a user is a member of a public group, so we requery
+      var q2 = Group.findOne({ _id: group._id, privacy:'public', members: {$in:[user._id]} });
+      q2.exec(function(err, group2){
+        if (err) return next(err);
+        if (!group2 && userInGroup === undefined) { group.members.push(user); }
```

Comment:
Definitively, the check should be for `group2` not `!group2`. @TheHollidayInn sorry for all the requests but could you fix this and add a test that checks that if you're seeing a public guild you're not member of you don't get added to its member list? Thanks